Sprint 16 (home work) Gr26

amanzhola opened the pull request.

amanzhola committed:
Sprint 15 (home work) Gr26
RU (по умолчанию):
a) 🧢🔒 — тулбар (но при нажатии: 🔴🎯)
b) 🧩✅ — main
c) 📝🚫😊 — строки

EN:
a) 🧢🔓🔴 — тулбар, (dropdown, filter)
b) 🧩❌➡️😄 — в main
c) 📝🔁❤️😄 — в строках

amanzhola committed:
Sprint 15 (home work) Gr26
Благодаря "хлебному месту" Леонида Сухинина в группе Alexander Hamilton которая оказалось похлеще Красного Хлеба так что чуть пришлось подредактировать не только sw320dp и sw360dp, а даже sw393dp

amanzhola committed:
Sprint 15 (home work) Gr26
🧩 BaseActivity 💥 ➡️ (👨‍🔧 ➕ 🧰)

amanzhola committed:
Sprint 15 (home work) Gr26
🧩 BaseActivity 🔚 🤔 🧼 🧹 🏗
🧩 SettingsActivity 🛠 ➡️ ♻️ 🧼 🏗
🧩 MainActivity✳️ 🎯 ➡️ ♻️ 🧼 🏗

amanzhola committed:
Sprint 15 (home work) Gr26
change or add: 65 files
🧩 SearchActivity ♻️
🧩 SearchViewModel ♻️
🧩 SearchMovie ♻️
🧩 ExtraOption ♻️
🧩 ExtraOptionViewModel ♻️
🧩 LauncherActivity ♻️
🧩 MoviePager ♻️
🧩 MoviePagerList ♻️
🧩 MoviesAdapterList ♻️
🧩 TrackAdapter ♻️
🧩 TrackDetailActivity ♻️
🧩 TrackPreviewActivity ♻️
🧩 TrackPreviewViewModel ♻️

✳️ 🎯 ➡️ ♻️ 🧼 🏗
xml -> activity_movie_pager.xml, activity_search.xml, i_movie1.xml, move_down_24.xml, strings.xml, strings(en).xml,themes.xml

✳️ 🎯 ➡️ ♻️ 🧼 🏗
Agreement.kt, AgreementImpl, App.kt, AppShareHelper.kt, AudioErrorManager.kt, AudioSingleTrackImpl, AudioSingleTrackShare.kt, AudioTracksImpl.kt, AudioTracksShare.kt, BottomNavigationHelper.kt, CheckInternetConnectionUseCase.kt, ColorApplierHelper.kt, Creator.kt, GsonMovieSerializer.kt, GsonTrackSerializer.kt, JsonFileWriter.kt, LegalHelper.kt, MovieSerializer.kt, MovieStorageHelper.kt, NetworkChecker.kt, NetworkRepository.kt, NetworkRepositoryImpl, NetworkStatusChecker.kt, NetworkStatusCheckerImpl, NetworkUtils.kt, ResourceColorProvider.kt, ResourceColorProviderImpl, SearchUiState.kt, SegmentManager.kt, SegmentTextHelper.kt, Share.kt, SharedPrefsMovieStorage.kt, SharedPrefsTrackStorage.kt, ShareImpl.kt, ShareMovie.kt, ShareMovieImpl.kt, Support.kt, SupportHelper.kt, SupportImpl.kt, TrackListInputData.kt, TrackListIntentParser.kt, TrackListIntentParser.kt, TrackListIntentParserImpl.kt, TrackSerializer.kt, TrackStorageHelper.kt

➕ 🎥➡️⬅️🎥 ➡️ (🆕 🎬 Обмен фильмами доступен )
➕ 🌺🔄 🌼🔃 ➡️ (✨ 🌸 Цветок на тулбаре теперь вращается)

🔚 🤔 🧼 🧹 🏗

amanzhola committed:
Sprint 15 (home work) Gr26
🔧✅✏️📴🔻🆙🔘(fixed update btn underline)

amanzhola committed:
Sprint 15 (home work) Gr26
⭐ 📽️ 🍿 🎬 ✨ 💖 🔝 (Movie 💖 done)
+ tx 🚀🔥 Vlamidir(🐞➖ ➡️ 🚫 ➡️ 😎)

amanzhola committed:
Sprint 16 (home work) Gr26

amanzhola committed:
Sprint 15 (home work) Gr26
(reserve step)

anastasianadelshtehel commented on app/src/main/java/com/example/playlistmaker/App.kt:
```diff
@@ -1,19 +1,33 @@
+import com.example.playlistmaker.domain.usecases.base.ThemeManager
+import com.example.playlistmaker.presentation.utils.ThemeLanguageHelper
+
 // ☀️ 🔁 🌙 👉 🧼🏗️✅
```
Комментарии в виде эмоджи не должны использоваться в коде
На реальном проекте не должно быть комментарием, кроме документации

anastasianadelshtehel commented on app/src/main/java/com/example/playlistmaker/presentation/launcherViewModels/TrackPreviewViewModel.kt:
Рекомендация: Объединить в единый класс состояния

anastasianadelshtehel commented on app/src/main/java/com/example/playlistmaker/ui/audio/SearchActivity.kt:
```diff
@@ -1,129 +1,107 @@
     private lateinit var viewModel: SearchViewModel
-    private lateinit var debounce: Debounce
-
+    private val debounce = Debounce(SEARCH_DEBOUNCE_DELAY)
```
Проблемы с жизненным циклом. Лучше перенести в ViewModel

anastasianadelshtehel reviewed (commented):
баг: приложение не запускается
![image](https://github.com/user-attachments/assets/934059c4-d8f3-422d-a995-45b77013d3ea)

amanzhola committed:
Sprint 16 (home work) Gr26
1) emoji - мне 54 -> 0 шансов быть устроиться и быть реальном проекте кроме своего проекта: status - без измненения
2) "баг" - не имеет отношения к классу LauncherActivity. Убрал из манифеста 2 старта .Main и .LAUNCHER поэтому у вас возникает "баг", а у меня нет
+ заменил TrackPreviewActivity provider через Creator
3) Чип и Дейл в лице Ивана Бартцева спешат на помощь

amanzhola committed:
Sprint 16 (home work) Gr26
1) emoji - мне 54 -> 0 шансов быть устроиться и быть реальном проекте кроме своего проекта: status - без измненения
2) "баг" - не имеет отношения к классу LauncherActivity. Убрал из манифеста 2 старта .Main и .LAUNCHER поэтому у вас возникает "баг", а у меня нет
+ заменил TrackPreviewActivity provider через Creator
3) Чип и Дейл в лице Ивана Бартцева спешат на помощь

amanzhola committed:
Sprint 16 (home work) Gr26
1) emoji -> мне 54 и 0 шансов устроиться и быть в реальном проекте кроме своего проекта: status - без измненения
2) "баг" -> был не в классе LauncherActivity Убрал из манифеста 2 старта .Main и .LAUNCHER, + Иван Бартцев (наставник) показал: надо запуск делать из app, у меня было из main, поэтому у вас есть "баг", а у меня не было
+ заменил TrackPreviewActivity provider через Creator
3) Здесь уже ИИ запихал в стейты, ранее не получалось так как мерцал или моргал экран откуда родилась TrackAdapterAudio(дифтул другой) и изменения через поля модели в 14. Позже в 16 уже практикум ввел кино isFavorite (реализовано в кино)
4) reactive cleaning track's list once History cleaned -> done by SearchViewModel

И спасибо за TrackPreviewActivity и ExtraOption. Теперь самотек на двоих один viewmodel как сейчас адаптер (буду делать). Вообще TrackActivityPreview родилась как временное. Текущий ExtraOption умеет отличать и дает один вид если из Search и другой если не оттуда и в оригинале на ExtraOption должно быть сокрытие поля ввода текста если из Launcher и наоброт если из Search, но пока выжидаю из-за незнания по какому пути пойдет практикум. Поэтому пока SearchHistoryInteraction дает временную жизнь для TrackStorageHelper не убивая

amanzhola committed:
Sprint 16 (home work) Gr26
5) спс Анастасии: избегал думая viewmodel потребует зависимости через фабрику для debounce, оказалось ошибочное представление
Updates:
SearchActivity -> - debounce
SearchViewModel -> + debounce
SearchMovie -> - debounce
MoviesViewModel -> + debounce
SearchWeather -> no viewmodel, no update
